File: test/system/single_line_test.rb
require "application_system_test_case"

class SingleLineTest < ApplicationSystemTestCase
  test "disable multi-line" do
    visit edit_post_path(posts(:empty), multi_line_disabled: true)

    find_editor.send "Hello"
    find_editor.send :enter
    find_editor.send "there"

    assert_editor_html "<p>Hellothere</p>"
  end
end

File: test/browser/tests/single_line.test.js
import { test } from "../test_helper.js"
import { assertEditorHtml } from "../helpers/assertions.js"

test.describe("Single line", () => {
  test("disable multi-line", async ({ page, editor }) => {
    await page.goto("/posts/new?multi_line_disabled=true")
    await editor.waitForConnected()

    await editor.send("Hello")
    await editor.send("Enter")
    await editor.send("there")

    await assertEditorHtml(editor, "<p>Hellothere</p>")
  })
})
